x-pack/plugins/fleet/server/services/epm/archive/cache.ts

```diff
@@ -5,23 +5,37 @@
  * 2.0.
  */
 
-import type { ArchivePackage, RegistryPackage } from '../../../../common';
+import type { ArchivePackage, RegistryPackage, PackageVerificationResult } from '../../../types';
 import { appContextService } from '../..';
 
 import type { ArchiveEntry } from '.';
 
+type SharedKeyString = string;
+
+const sharedKey = ({ name, version }: SharedKey): SharedKeyString => `${name}-${version}`;
+
 const archiveEntryCache: Map<ArchiveEntry['path'], ArchiveEntry['buffer']> = new Map();
 export const getArchiveEntry = (key: string) => archiveEntryCache.get(key);
 export const setArchiveEntry = (key: string, value: Buffer) => archiveEntryCache.set(key, value);
 export const hasArchiveEntry = (key: string) => archiveEntryCache.has(key);
 export const clearArchiveEntries = () => archiveEntryCache.clear();
 export const deleteArchiveEntry = (key: string) => archiveEntryCache.delete(key);
 
+const verificationResultCache: Map<SharedKeyString, PackageVerificationResult> = new Map();
```

Comment:
I am wondering if we should avoid this cache pattern at some point it's not going to scale (it's probably more problematic for `archiveEntryCache`) do we really need to cache things how often are computing the verification? 

If we really need to cache it it's probably better to use an LRU cache with a max size